neossat2caom2/tests/test_composable.py

```diff
@@ -104,90 +102,117 @@
+    
+    class MockDS(HttpDataSource):
+
+        def __init__(self, config, start_key, html_filters, session):
+            super().__init__(config, start_key, html_filters, session)
+
+        def _descend_html_hierarchy(self, ignore_node):
+            self._todo_list = x
+
+    data_source_mock.side_effect = MockDS
     access_mock.return_value = 'https://localhost'
 
     test_config.change_working_directory(tmpdir)
     test_config.task_types = [mc.TaskType.STORE, mc.TaskType.INGEST]
     test_config.proxy_file_name = 'textproxy.pem'
     test_config.interval = 200
-    test_config.write_to_file(test_config)
-
-    mc.State.write_bookmark(test_config.state_fqn, test_config.bookmark, TEST_START_TIME)
-    with open(test_config.proxy_fqn, 'w') as f:
-        f.write('test content')
+    test_config.logging_level = 'INFO'
 
-    run_mock.return_value = 0
+    orig_dir = os.getcwd()
     try:
-        test_result = composable._run_state()
-    except Exception as e:
-        import traceback
-        import logging
-        logging.error(traceback.format_exc())
-        assert False, f'unexpected exception {e}'
-    assert test_result == 0, 'expected successful test result'
-    assert initialize_mock.called, 'initialize should be called'
-    # called for every invocation of get_time_box_work
-    assert initialize_mock.call_count == 12, 'wrong initialize call count'
-    assert run_mock.called, 'should be called'
-    args, kwargs = run_mock.call_args
-    test_storage = args[0]
-    assert isinstance(test_storage, NEOSSatName), type(test_storage)
-    assert test_storage.file_name.startswith(
-        'NEOS_SCI'
-    ) and test_storage.file_name.endswith('.fits'), test_storage.file_name
-    assert run_mock.call_count == 10, 'wrong call count'
-    test_state_end = mc.State(test_config.state_fqn, test_config.time_zone)
-    test_end_bookmark = test_state_end.get_bookmark(test_config.bookmark)
-    assert TEST_DT_2 == test_end_bookmark, f'wrong end time max {TEST_DT_2} end {test_end_bookmark}'
+        os.chdir(tmpdir)
+ 
+        test_config.write_to_file(test_config)
+        mc.State.write_bookmark(test_config.state_fqn, test_config.data_sources[0], TEST_START_TIME)
+        with open(test_config.proxy_fqn, 'w') as f:
+            f.write('test content')
+
+        run_mock.return_value = 0
+        try:
+            test_result = composable._run_state()
+        except Exception as e:
+            logging.error(traceback.format_exc())
+            assert False, f'unexpected exception {e}'
+        assert test_result == 0, 'expected successful test result'
+        assert run_mock.called, 'should be called'
+        args, kwargs = run_mock.call_args
+        test_storage = args[0]
+        assert isinstance(test_storage, NEOSSatName), type(test_storage)
+        assert test_storage.file_name.startswith(
+            'NEOS_SCI'
+        ) and test_storage.file_name.endswith('.fits'), test_storage.file_name
```

Comment:
If you don't mind regex (`re` needs imported`)
```suggestion
        assert re.match('^NEOS_SCI.*\.fits$', test_storage.file_name) is not None
```

Reply:
Done.